File: libs/python/BlueJeansMeetingsRestApi/models/personal_meeting_room_advanced_meeting_options_editability.py
# coding: utf-8

"""
    BlueJeans Meetings REST API

     ## Video That Works Where You Do. This site provides developers access to APIs from BlueJean's Meetings meeting service.  From here you will be able to make actual API calls to manage User Accounts, Meetings, and Recordings.  Also, you can pull analytical data and current state information. With these APIs you can quickly integrate **BlueJeans** video administration into your applications.    <alert>7/010/2018 </alert>Deprecated some Command Center API's.  Exposed API's that Summarize usage.<br/> 7/06/2018 Added API for changing ownership of recordings.<br/> 6/28/2018 Restored Create Enterprise Account, added Enterprise and User Profile Tag API's.<br/> 6/01/2018 Clarified Access Token requirements for meeting endpoint API's.<br/> 5/21/2018 Corrected return model for meeting history call.<br/> <hr>  <h2 class=\"info\"> What is a BlueJeans Meeting?</h2> <img src=\"./img/bjnmeeting.png\" style=\"padding-left:20px; width:50%\">  A BlueJeans meeting is a collaboration session of 2 or more participants joining a virtual meeting-room in the cloud.   The first person to join is put into a waiting-room until other participant(s) join.  When the meeting starts, all participants will be connected over video & audio.  <u>Types of Meeting Rooms</u> There are two types of Meeting Rooms available to a registered user.  They are described as: <ul> <li>Scheduled – A room created for a specific date and time.  It is assigned a randomized and unique numeric ID by BlueJeans.</li> <li>Personal – Every user is given a personal meeting room with a customized Meeting Identifier.  People typically use their mobile phone number as their Meeting ID.</li> </ul> <u>Meeting Room Security</u> A meeting room can <i>optionally</i> be protected by a randomized access code.  THe access code ensures security of participants who join.  Also the access code can specify a Moderator role whose user(s) have additional capabilities to administer a meeting. <ul> <li>When set, participants would be required to enter the access code in order to join.</li> <li>The scheduler's profile has a user-specified moderator passcode which can be input to allow designated user(s) greater privileges to help organize meetings.</li> </ul> <hr>  <h2 id='gs' class=\"info\" onclick=\"hideit('gs')\"><span id=\"gsicon\" class=\"glyphicon glyphicon-chevron-right\"></span> Getting Started</h2> <div id=\"gsbody\" style=\"display:none\"> Before you start using BlueJeans' APIs, you must first have a BlueJeans account enabled for API Access.  Contact <a href=\"mailto:Support@BlueJeans.com)\">BlueJeans Support</a> for assistance.  <br /><br />Once you have an account, you may start writing application code to authenticate and make API calls.  *Alternatively*, you can use this developer site to test the BlueJeans' APIs and develop a level of familiarity before you write production code.  <br />  ### To Make API Calls from This Site If you want to use this developer site to try various BlueJeans' APIs, here are the steps required to authenticate and enable your Developer Session to place API calls. 1. Choose Method for Authenticating      * Click on the desired Authentication method from below.      * Click on the **Try It Out** button. 1. Make Authentication request      * Follow APIs instructions and input the API parameters.      * Click on the blue **Execute** button.      * If successful, the API returns with JSON data containing a field called **access_token**.  Copy/save this value. 1. Authorize BlueJeans Developer Session.      * Click on the green **Authorize button**.       * The site will show you a pop-up window for authorization.      * Enter your access token in the field named **api_key**      * Click on the **Authorize** button  Your current BlueJeans developer session is now authenticated and ready to place API calls.  The web site will automatically include your access token on any API calls you make. </div> <hr>  <h2 id='oauth' class=\"info\" onclick=\"hideit('oauth')\"><span id=\"oauthicon\" class=\"glyphicon glyphicon-chevron-right\"></span> About Meetings Authentication</h2>  <div id=\"oauthbody\" style=\"display:none\">  All API transactions (excluding Authentication) require an access token per **OAuth standards**.  BlueJeans provides multiple methods for obtaining an access token.  Additionally there are diffferent scopes of token access. ### Grant Types Bluejeans provides 4 different methods for users to Authenticate.  Successful authentication allows BlueJeans to grant an access token to perform API operations. * Password Credentials Grant – Authenticate with a username and password and receive an access token with user level permission. Known as two-legged OAuth. * Meeting Credentials Grant – Authenticate with a meeting ID and meeting passcode and receive an access token with meeting level permission. Known as two-legged OAuth. * Client Credentials Grant –  Authenticate with a client ID and client secret and receive an access token with enterprise level permission. Known as two-legged OAuth. * Authorization Code Grant – Authentication for your developer's application occurs through a redirection to a BlueJeans authentication page. The application receives an authorization code to be submitted, along with other tokens, to receive an access token. Known as three-legged OAuth. For more information please refer to the [OAuth specification](https://oauth.net/). ### Access & Permissions BlueJeans defines 3 levels of API access into the system.  When an access token is granted, it carries one of these 3 levels.  The scope of system functionality depends upon the token's access level. * Meeting-level – Scope of APIs is limited to individual meetings. * User-level – Scope depends on the requested permissions. * App-level – provisioned either by BlueJeans personnel, or the BlueJeans Enterprise Admin, an app, is issued a client key and secret key. These tokens then are used by the BlueJeans Authentication API to receive the token. The token's scope provides access to the entire enterprise and all of its users. All endpoints in this document that require **Enterprise Admin** access will be marked as such. </div> <hr> 

    OpenAPI spec version: 1.0.4407102018
    Contact: glenn@bluejeans.com
    Generated by: https://github.com/swagger-api/swagger-codegen.git
"""


from pprint import pformat
from six import iteritems
import re


class PersonalMeetingRoomAdvancedMeetingOptionsEditability(object):
    """
    NOTE: This class is auto generated by the swagger code generator program.
    Do not edit the class manually.
    """


    """
    Attributes:
      swagger_types (dict): The key is attribute name
                            and the value is attribute type.
      attribute_map (dict): The key is attribute name
                            and the value is json key in definition.
    """
    swagger_types = {
        'auto_record': 'bool',
        'enforce_meeting_encryption': 'bool',
        'video_best_fit': 'bool',
        'enforce_meeting_encryption_allow_pstn': 'bool',
        'disallow_chat': 'bool',
        'mute_participants_on_entry': 'bool',
        'moderator_less': 'bool',
        'show_all_attendees_in_meeting_invite': 'bool'
    }

    attribute_map = {
        'auto_record': 'autoRecord',
        'enforce_meeting_encryption': 'enforceMeetingEncryption',
        'video_best_fit': 'videoBestFit',
        'enforce_meeting_encryption_allow_pstn': 'enforceMeetingEncryptionAllowPSTN',
        'disallow_chat': 'disallowChat',
        'mute_participants_on_entry': 'muteParticipantsOnEntry',
        'moderator_less': 'moderatorLess',
        'show_all_attendees_in_meeting_invite': 'showAllAttendeesInMeetingInvite'
    }

    def __init__(self, auto_record=None, enforce_meeting_encryption=None, video_best_fit=None, enforce_meeting_encryption_allow_pstn=None, disallow_chat=None, mute_participants_on_entry=None, moderator_less=None, show_all_attendees_in_meeting_invite=None):
        """
        PersonalMeetingRoomAdvancedMeetingOptionsEditability - a model defined in Swagger
        """

        self._auto_record = None
        self._enforce_meeting_encryption = None
        self._video_best_fit = None
        self._enforce_meeting_encryption_allow_pstn = None
        self._disallow_chat = None
        self._mute_participants_on_entry = None
        self._moderator_less = None
        self._show_all_attendees_in_meeting_invite = None

        if auto_record is not None:
          self.auto_record = auto_record
        if enforce_meeting_encryption is not None:
          self.enforce_meeting_encryption = enforce_meeting_encryption
        if video_best_fit is not None:
          self.video_best_fit = video_best_fit
        if enforce_meeting_encryption_allow_pstn is not None:
          self.enforce_meeting_encryption_allow_pstn = enforce_meeting_encryption_allow_pstn
        if disallow_chat is not None:
          self.disallow_chat = disallow_chat
        if mute_participants_on_entry is not None:
          self.mute_participants_on_entry = mute_participants_on_entry
        if moderator_less is not None:
          self.moderator_less = moderator_less
        if show_all_attendees_in_meeting_invite is not None:
          self.show_all_attendees_in_meeting_invite = show_all_attendees_in_meeting_invite

    @property
    def auto_record(self):
        """
        Gets the auto_record of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The auto_record of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._auto_record

    @auto_record.setter
    def auto_record(self, auto_record):
        """
        Sets the auto_record of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param auto_record: The auto_record of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._auto_record = auto_record

    @property
    def enforce_meeting_encryption(self):
        """
        Gets the enforce_meeting_encryption of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The enforce_meeting_encryption of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._enforce_meeting_encryption

    @enforce_meeting_encryption.setter
    def enforce_meeting_encryption(self, enforce_meeting_encryption):
        """
        Sets the enforce_meeting_encryption of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param enforce_meeting_encryption: The enforce_meeting_encryption of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._enforce_meeting_encryption = enforce_meeting_encryption

    @property
    def video_best_fit(self):
        """
        Gets the video_best_fit of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The video_best_fit of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._video_best_fit

    @video_best_fit.setter
    def video_best_fit(self, video_best_fit):
        """
        Sets the video_best_fit of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param video_best_fit: The video_best_fit of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._video_best_fit = video_best_fit

    @property
    def enforce_meeting_encryption_allow_pstn(self):
        """
        Gets the enforce_meeting_encryption_allow_pstn of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The enforce_meeting_encryption_allow_pstn of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._enforce_meeting_encryption_allow_pstn

    @enforce_meeting_encryption_allow_pstn.setter
    def enforce_meeting_encryption_allow_pstn(self, enforce_meeting_encryption_allow_pstn):
        """
        Sets the enforce_meeting_encryption_allow_pstn of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param enforce_meeting_encryption_allow_pstn: The enforce_meeting_encryption_allow_pstn of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._enforce_meeting_encryption_allow_pstn = enforce_meeting_encryption_allow_pstn

    @property
    def disallow_chat(self):
        """
        Gets the disallow_chat of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The disallow_chat of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._disallow_chat

    @disallow_chat.setter
    def disallow_chat(self, disallow_chat):
        """
        Sets the disallow_chat of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param disallow_chat: The disallow_chat of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._disallow_chat = disallow_chat

    @property
    def mute_participants_on_entry(self):
        """
        Gets the mute_participants_on_entry of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The mute_participants_on_entry of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._mute_participants_on_entry

    @mute_participants_on_entry.setter
    def mute_participants_on_entry(self, mute_participants_on_entry):
        """
        Sets the mute_participants_on_entry of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param mute_participants_on_entry: The mute_participants_on_entry of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._mute_participants_on_entry = mute_participants_on_entry

    @property
    def moderator_less(self):
        """
        Gets the moderator_less of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The moderator_less of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._moderator_less

    @moderator_less.setter
    def moderator_less(self, moderator_less):
        """
        Sets the moderator_less of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param moderator_less: The moderator_less of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._moderator_less = moderator_less

    @property
    def show_all_attendees_in_meeting_invite(self):
        """
        Gets the show_all_attendees_in_meeting_invite of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :return: The show_all_attendees_in_meeting_invite of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :rtype: bool
        """
        return self._show_all_attendees_in_meeting_invite

    @show_all_attendees_in_meeting_invite.setter
    def show_all_attendees_in_meeting_invite(self, show_all_attendees_in_meeting_invite):
        """
        Sets the show_all_attendees_in_meeting_invite of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        null

        :param show_all_attendees_in_meeting_invite: The show_all_attendees_in_meeting_invite of this PersonalMeetingRoomAdvancedMeetingOptionsEditability.
        :type: bool
        """

        self._show_all_attendees_in_meeting_invite = show_all_attendees_in_meeting_invite

    def to_dict(self):
        """
        Returns the model properties as a dict
        """
        result = {}

        for attr, _ in iteritems(self.swagger_types):
            value = getattr(self, attr)
            if isinstance(value, list):
                result[attr] = list(map(
                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
                    value
                ))
            elif hasattr(value, "to_dict"):
                result[attr] = value.to_dict()
            elif isinstance(value, dict):
                result[attr] = dict(map(
                    lambda item: (item[0], item[1].to_dict())
                    if hasattr(item[1], "to_dict") else item,
                    value.items()
                ))
            else:
                result[attr] = value

        return result

    def to_str(self):
        """
        Returns the string representation of the model
        """
        return pformat(self.to_dict())

    def __repr__(self):
        """
        For `print` and `pprint`
        """
        return self.to_str()

    def __eq__(self, other):
        """
        Returns true if both objects are equal
        """
        if not isinstance(other, PersonalMeetingRoomAdvancedMeetingOptionsEditability):
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        """
        Returns true if both objects are not equal
        """
        return not self == other

File: libs/csharp/src/com.bluejeans.api.rest.meetings/Model/PersonalMeetingRoomAdvancedMeetingOptionsEditability.cs
/* 
 * BlueJeans Meetings REST API
 *
 *  ## Video That Works Where You Do. This site provides developers access to APIs from BlueJean's Meetings meeting service.  From here you will be able to make actual API calls to manage User Accounts, Meetings, and Recordings.  Also, you can pull analytical data and current state information. With these APIs you can quickly integrate **BlueJeans** video administration into your applications.    <alert>7/23/2018 </alert>Corrected errors in API definition file.<br/> 7/10/2018 Deprecated some Command Center API's.  Exposed API's that Summarize usage.<br/> 7/06/2018 Added API for changing ownership of recordings.<br/> 6/28/2018 Restored Create Enterprise Account, added Enterprise and User Profile Tag API's.<br/> 6/01/2018 Clarified Access Token requirements for meeting endpoint API's.<br/> 5/21/2018 Corrected return model for meeting history call.<br/> <hr>  <h2 class=\"info\"> What is a BlueJeans Meeting?</h2> <img src=\"./img/bjnmeeting.png\" style=\"padding-left:20px; width:50%\">  A BlueJeans meeting is a collaboration session of 2 or more participants joining a virtual meeting-room in the cloud.   The first person to join is put into a waiting-room until other participant(s) join.  When the meeting starts, all participants will be connected over video & audio.  <u>Types of Meeting Rooms</u> There are two types of Meeting Rooms available to a registered user.  They are described as: <ul> <li>Scheduled – A room created for a specific date and time.  It is assigned a randomized and unique numeric ID by BlueJeans.</li> <li>Personal – Every user is given a personal meeting room with a customized Meeting Identifier.  People typically use their mobile phone number as their Meeting ID.</li> </ul> <u>Meeting Room Security</u> A meeting room can <i>optionally</i> be protected by a randomized access code.  THe access code ensures security of participants who join.  Also the access code can specify a Moderator role whose user(s) have additional capabilities to administer a meeting. <ul> <li>When set, participants would be required to enter the access code in order to join.</li> <li>The scheduler's profile has a user-specified moderator passcode which can be input to allow designated user(s) greater privileges to help organize meetings.</li> </ul> <hr>  <h2 id='gs' class=\"info\" onclick=\"hideit('gs')\"><span id=\"gsicon\" class=\"glyphicon glyphicon-chevron-right\"></span> Getting Started</h2> <div id=\"gsbody\" style=\"display:none\"> Before you start using BlueJeans' APIs, you must first have a BlueJeans account enabled for API Access.  Contact <a href=\"mailto:Support@BlueJeans.com)\">BlueJeans Support</a> for assistance.  <br /><br />Once you have an account, you may start writing application code to authenticate and make API calls.  *Alternatively*, you can use this developer site to test the BlueJeans' APIs and develop a level of familiarity before you write production code.  <br />  ### To Make API Calls from This Site If you want to use this developer site to try various BlueJeans' APIs, here are the steps required to authenticate and enable your Developer Session to place API calls. 1. Choose Method for Authenticating      * Click on the desired Authentication method from below.      * Click on the **Try It Out** button. 1. Make Authentication request      * Follow APIs instructions and input the API parameters.      * Click on the blue **Execute** button.      * If successful, the API returns with JSON data containing a field called **access_token**.  Copy/save this value. 1. Authorize BlueJeans Developer Session.      * Click on the green **Authorize button**.       * The site will show you a pop-up window for authorization.      * Enter your access token in the field named **api_key**      * Click on the **Authorize** button  Your current BlueJeans developer session is now authenticated and ready to place API calls.  The web site will automatically include your access token on any API calls you make. </div> <hr>  <h2 id='oauth' class=\"info\" onclick=\"hideit('oauth')\"><span id=\"oauthicon\" class=\"glyphicon glyphicon-chevron-right\"></span> About Meetings Authentication</h2>  <div id=\"oauthbody\" style=\"display:none\">  All API transactions (excluding Authentication) require an access token per **OAuth standards**.  BlueJeans provides multiple methods for obtaining an access token.  Additionally there are diffferent scopes of token access. ### Grant Types Bluejeans provides 4 different methods for users to Authenticate.  Successful authentication allows BlueJeans to grant an access token to perform API operations. * Password Credentials Grant – Authenticate with a username and password and receive an access token with user level permission. Known as two-legged OAuth. * Meeting Credentials Grant – Authenticate with a meeting ID and meeting passcode and receive an access token with meeting level permission. Known as two-legged OAuth. * Client Credentials Grant –  Authenticate with a client ID and client secret and receive an access token with enterprise level permission. Known as two-legged OAuth. * Authorization Code Grant – Authentication for your developer's application occurs through a redirection to a BlueJeans authentication page. The application receives an authorization code to be submitted, along with other tokens, to receive an access token. Known as three-legged OAuth. For more information please refer to the [OAuth specification](https://oauth.net/). ### Access & Permissions BlueJeans defines 3 levels of API access into the system.  When an access token is granted, it carries one of these 3 levels.  The scope of system functionality depends upon the token's access level. * Meeting-level – Scope of APIs is limited to individual meetings. * User-level – Scope depends on the requested permissions. * App-level – provisioned either by BlueJeans personnel, or the BlueJeans Enterprise Admin, an app, is issued a client key and secret key. These tokens then are used by the BlueJeans Authentication API to receive the token. The token's scope provides access to the entire enterprise and all of its users. All endpoints in this document that require **Enterprise Admin** access will be marked as such. </div> <hr> 
 *
 * OpenAPI spec version: 1.0.4407232018
 * Contact: glenn@bluejeans.com
 * Generated by: https://github.com/swagger-api/swagger-codegen.git
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;
using SwaggerDateConverter = com.bluejeans.api.rest.meetings.Client.SwaggerDateConverter;

namespace com.bluejeans.api.rest.meetings.Model
{
    /// <summary>
    /// PersonalMeetingRoomAdvancedMeetingOptionsEditability
    /// </summary>
    [DataContract]
    public partial class PersonalMeetingRoomAdvancedMeetingOptionsEditability :  IEquatable<PersonalMeetingRoomAdvancedMeetingOptionsEditability>, IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="PersonalMeetingRoomAdvancedMeetingOptionsEditability" /> class.
        /// </summary>
        /// <param name="autoRecord">Allow UI to edit AutoRecord field.</param>
        /// <param name="enforceMeetingEncryption">Allow UI to edit this field.</param>
        /// <param name="videoBestFit">Allow UI to edict videoBestFit field.</param>
        /// <param name="enforceMeetingEncryptionAllowPSTN">Allow UI to edit this field.</param>
        /// <param name="disallowChat">Allow UI to edit the disallowChat field.</param>
        /// <param name="muteParticipantsOnEntry">Allow UI to edit muteParticipantsOnEntry field.</param>
        /// <param name="moderatorLess">Allow UI to edit moderatorLess field.</param>
        /// <param name="showAllAttendeesInMeetingInvite">Allow UI to edit this field.</param>
        public PersonalMeetingRoomAdvancedMeetingOptionsEditability(bool? autoRecord = default(bool?), bool? enforceMeetingEncryption = default(bool?), bool? videoBestFit = default(bool?), bool? enforceMeetingEncryptionAllowPSTN = default(bool?), bool? disallowChat = default(bool?), bool? muteParticipantsOnEntry = default(bool?), bool? moderatorLess = default(bool?), bool? showAllAttendeesInMeetingInvite = default(bool?))
        {
            this.AutoRecord = autoRecord;
            this.EnforceMeetingEncryption = enforceMeetingEncryption;
            this.VideoBestFit = videoBestFit;
            this.EnforceMeetingEncryptionAllowPSTN = enforceMeetingEncryptionAllowPSTN;
            this.DisallowChat = disallowChat;
            this.MuteParticipantsOnEntry = muteParticipantsOnEntry;
            this.ModeratorLess = moderatorLess;
            this.ShowAllAttendeesInMeetingInvite = showAllAttendeesInMeetingInvite;
        }
        
        /// <summary>
        /// Allow UI to edit AutoRecord field
        /// </summary>
        /// <value>Allow UI to edit AutoRecord field</value>
        [DataMember(Name="autoRecord", EmitDefaultValue=false)]
        public bool? AutoRecord { get; set; }

        /// <summary>
        /// Allow UI to edit this field
        /// </summary>
        /// <value>Allow UI to edit this field</value>
        [DataMember(Name="enforceMeetingEncryption", EmitDefaultValue=false)]
        public bool? EnforceMeetingEncryption { get; set; }

        /// <summary>
        /// Allow UI to edict videoBestFit field
        /// </summary>
        /// <value>Allow UI to edict videoBestFit field</value>
        [DataMember(Name="videoBestFit", EmitDefaultValue=false)]
        public bool? VideoBestFit { get; set; }

        /// <summary>
        /// Allow UI to edit this field
        /// </summary>
        /// <value>Allow UI to edit this field</value>
        [DataMember(Name="enforceMeetingEncryptionAllowPSTN", EmitDefaultValue=false)]
        public bool? EnforceMeetingEncryptionAllowPSTN { get; set; }

        /// <summary>
        /// Allow UI to edit the disallowChat field
        /// </summary>
        /// <value>Allow UI to edit the disallowChat field</value>
        [DataMember(Name="disallowChat", EmitDefaultValue=false)]
        public bool? DisallowChat { get; set; }

        /// <summary>
        /// Allow UI to edit muteParticipantsOnEntry field
        /// </summary>
        /// <value>Allow UI to edit muteParticipantsOnEntry field</value>
        [DataMember(Name="muteParticipantsOnEntry", EmitDefaultValue=false)]
        public bool? MuteParticipantsOnEntry { get; set; }

        /// <summary>
        /// Allow UI to edit moderatorLess field
        /// </summary>
        /// <value>Allow UI to edit moderatorLess field</value>
        [DataMember(Name="moderatorLess", EmitDefaultValue=false)]
        public bool? ModeratorLess { get; set; }

        /// <summary>
        /// Allow UI to edit this field
        /// </summary>
        /// <value>Allow UI to edit this field</value>
        [DataMember(Name="showAllAttendeesInMeetingInvite", EmitDefaultValue=false)]
        public bool? ShowAllAttendeesInMeetingInvite { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class PersonalMeetingRoomAdvancedMeetingOptionsEditability {\n");
            sb.Append("  AutoRecord: ").Append(AutoRecord).Append("\n");
            sb.Append("  EnforceMeetingEncryption: ").Append(EnforceMeetingEncryption).Append("\n");
            sb.Append("  VideoBestFit: ").Append(VideoBestFit).Append("\n");
            sb.Append("  EnforceMeetingEncryptionAllowPSTN: ").Append(EnforceMeetingEncryptionAllowPSTN).Append("\n");
            sb.Append("  DisallowChat: ").Append(DisallowChat).Append("\n");
            sb.Append("  MuteParticipantsOnEntry: ").Append(MuteParticipantsOnEntry).Append("\n");
            sb.Append("  ModeratorLess: ").Append(ModeratorLess).Append("\n");
            sb.Append("  ShowAllAttendeesInMeetingInvite: ").Append(ShowAllAttendeesInMeetingInvite).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as PersonalMeetingRoomAdvancedMeetingOptionsEditability);
        }

        /// <summary>
        /// Returns true if PersonalMeetingRoomAdvancedMeetingOptionsEditability instances are equal
        /// </summary>
        /// <param name="input">Instance of PersonalMeetingRoomAdvancedMeetingOptionsEditability to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(PersonalMeetingRoomAdvancedMeetingOptionsEditability input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.AutoRecord == input.AutoRecord ||
                    (this.AutoRecord != null &&
                    this.AutoRecord.Equals(input.AutoRecord))
                ) && 
                (
                    this.EnforceMeetingEncryption == input.EnforceMeetingEncryption ||
                    (this.EnforceMeetingEncryption != null &&
                    this.EnforceMeetingEncryption.Equals(input.EnforceMeetingEncryption))
                ) && 
                (
                    this.VideoBestFit == input.VideoBestFit ||
                    (this.VideoBestFit != null &&
                    this.VideoBestFit.Equals(input.VideoBestFit))
                ) && 
                (
                    this.EnforceMeetingEncryptionAllowPSTN == input.EnforceMeetingEncryptionAllowPSTN ||
                    (this.EnforceMeetingEncryptionAllowPSTN != null &&
                    this.EnforceMeetingEncryptionAllowPSTN.Equals(input.EnforceMeetingEncryptionAllowPSTN))
                ) && 
                (
                    this.DisallowChat == input.DisallowChat ||
                    (this.DisallowChat != null &&
                    this.DisallowChat.Equals(input.DisallowChat))
                ) && 
                (
                    this.MuteParticipantsOnEntry == input.MuteParticipantsOnEntry ||
                    (this.MuteParticipantsOnEntry != null &&
                    this.MuteParticipantsOnEntry.Equals(input.MuteParticipantsOnEntry))
                ) && 
                (
                    this.ModeratorLess == input.ModeratorLess ||
                    (this.ModeratorLess != null &&
                    this.ModeratorLess.Equals(input.ModeratorLess))
                ) && 
                (
                    this.ShowAllAttendeesInMeetingInvite == input.ShowAllAttendeesInMeetingInvite ||
                    (this.ShowAllAttendeesInMeetingInvite != null &&
                    this.ShowAllAttendeesInMeetingInvite.Equals(input.ShowAllAttendeesInMeetingInvite))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.AutoRecord != null)
                    hashCode = hashCode * 59 + this.AutoRecord.GetHashCode();
                if (this.EnforceMeetingEncryption != null)
                    hashCode = hashCode * 59 + this.EnforceMeetingEncryption.GetHashCode();
                if (this.VideoBestFit != null)
                    hashCode = hashCode * 59 + this.VideoBestFit.GetHashCode();
                if (this.EnforceMeetingEncryptionAllowPSTN != null)
                    hashCode = hashCode * 59 + this.EnforceMeetingEncryptionAllowPSTN.GetHashCode();
                if (this.DisallowChat != null)
                    hashCode = hashCode * 59 + this.DisallowChat.GetHashCode();
                if (this.MuteParticipantsOnEntry != null)
                    hashCode = hashCode * 59 + this.MuteParticipantsOnEntry.GetHashCode();
                if (this.ModeratorLess != null)
                    hashCode = hashCode * 59 + this.ModeratorLess.GetHashCode();
                if (this.ShowAllAttendeesInMeetingInvite != null)
                    hashCode = hashCode * 59 + this.ShowAllAttendeesInMeetingInvite.GetHashCode();
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}
